### PR TITLE
[fix] polygamma n>=1

### LINKED ISSUE #55357: [bug] torch.polygamma inconsistent with scipy.special.polygamma for n >= 1
n = 1
```python
>>> t = torch.tensor([-1.])
>>> torch.polygamma(1, t)
tensor([1.2914e+15])
>>> scipy.special.polygamma(1, t.numpy())
array([inf], dtype=float32)

>>> t = torch.tensor([-501.])
>>> torch.polygamma(1, t)
tensor([2.0831e+09])
>>> scipy.special.polygamma(1, t.numpy())
array([inf], dtype=float32)

>>> t = torch.tensor([-float('inf')])
>>> torch.polygamma(1, t)
tensor([nan])
>>> scipy.special.polygamma(1, t.numpy())
array([inf], dtype=float32)
>>> 
```

n > 1
```python
>>> t = torch.tensor([float('inf')])
>>> torch.polygamma(2, t)
tensor([nan])
>>> scipy.special.polygamma(2, t.numpy())
array([-0.], dtype=float32)
>>> torch.polygamma(3, t)
tensor([nan])
>>> scipy.special.polygamma(3, t.numpy())
array([0.], dtype=float32)
>>> torch.polygamma(4, t)
tensor([nan])
>>> scipy.special.polygamma(4, t.numpy())
array([-0.], dtype=float32)
```

cc @mruberry

### PR DESCRIPTION
**Pull request by kshitij12345**
Fixes: #55357

TODO:
* [x] Use proper casting to avoid confusing the compiler

### CODE REVIEW

File: aten/src/ATen/native/Math.h [hunk 1]
```diff
@@ -251,42 +265,6 @@ C10_HOST_DEVICE static inline T polevl(const T x, const T A[], size_t len) {
   return result;
 }
 
-static inline double trigamma(double x) __ubsan_ignore_float_divide_by_zero__ {
```
Comment (kshitij12345) [inline aten/src/ATen/native/Math.h:254]: Remove the low precision variant: https://github.com/pytorch/pytorch/pull/4399

**Comment by kshitij12345** [comment]
Gentle ping @mruberry :)

**Comment by kshitij12345** [comment]
Gentle ping @mruberry :)

**Review by mruberry — APPROVED** [review]
Awesome! Great to have this fixed

**Comment by mruberry** [comment]
Doh! Looks like this picked up a merge conflict. Sorry about that @kshitij12345. Would you rebase it and ping me?

**Pull request closed without merging**